Mathlib/Algebra/GroupWithZero/Basic.lean

```diff
@@ -146,6 +147,16 @@ lemma zero_pow_eq (n : ℕ) : (0 : M₀) ^ n = if n = 0 then 1 else 0 := by
   · rw [h, pow_zero]
   · rw [zero_pow h]
 
+lemma eq_zero_of_zero_pow_eq_one₀ [Nontrivial M₀] : (0 : M₀) ^ n = 1 ↔ n = 0 := by
+  constructor
+  · intro ha
+    induction n with
+    | zero => rfl
+    | succ n _ => simp at ha
+  intro h0
+  rw[zero_pow_eq]
+  exact if_pos h0
```

Comment:
```suggestion
  rw [zero_pow_eq, zero_ne_one.ite_eq_left]
```

Reply:
Your suggestion doesn't seem to work.

Reply:
I guess what you meant was: `rw [zero_pow_eq, one_ne_zero.ite_eq_left_iff]`. At least this seems to work.